### DISCUSSION TITLE
How can I determine the units of concentration in the model output?

### DISCUSSION THREAD

**Discussion by Alvin-prc** [modflowpy/flopy#1942] [Q&A] [answered]
Thank you for taking the time to answer, I appreciate it.
1. In Flopy, is the unit of concentration in model output determined by the mass and length unit settings in flopy.mt3d.Mt3dBtn? 
2. For example, if munit="mg" and lunit="m", is the concentration unit mg/m3? Or is the concentration unit fixed? 
3. Is it necessary for the length and time units in flopy.mt3d.Mt3dBtn to match those set in flopy.modflow.ModflowDis?

**Comment by wpbonelli** [comment, marked as answer]
@Alvin-prc mf6 and related programs (including mt3dms & mt3d-usgs) do not generally prescribe units or use them for internal calculations/conversions. It is the user's responsibility to make sure input is given in consistent units. Flopy follows this convention. If the model grid discretization is given in meters, then it would be correct to interpret the unit volume in reported concentrations as m^3.

From the mt3d-usgs docs:

![mt3d-usgs-units](https://github.com/modflowpy/flopy/assets/10502868/b99a9048-d6e3-46f2-b20d-e68b62da6efc)

![mt3d-usgs-units-2](https://github.com/modflowpy/flopy/assets/10502868/e10e3ed5-38a9-43d8-b231-2a02cc991ab3)

**Comment by emorway-usgs** [comment]
@Alvin-prc, in addition to @w-bonelli's answer to your first question:
> In Flopy, is the unit of concentration in model output determined by the mass and length unit settings in flopy.mt3d.Mt3dBtn 

The unit specification helps with the labeling in the MT3DMS or MT3D-USGS standard output file.  Otherwise, as pointed out already, these parameters do not effect the simulation
